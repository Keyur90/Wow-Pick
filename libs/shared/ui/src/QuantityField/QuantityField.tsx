import React, { useState, useEffect, useRef } from 'react';
import TextField from '@material-ui/core/TextField';
import InputAdornment from '@material-ui/core/InputAdornment';
import { makeStyles } from '@material-ui/core/styles';

interface QuantityFieldPropTypes {
  name: string;
  label: string;
  defaultValue: number;
  onFocus?: () => void;
  onBlur?: (event: React.FocusEvent<HTMLInputElement>) => void;
  error: boolean;
  errorText?: string;
  shouldFocus?: boolean;
  startAdornment?: string;
}

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiInputBase-root': {
      marginBottom: theme.spacing(2.75),
      '&.Mui-error': {
        marginBottom: 0,
        '&.MuiOutlinedInput-colorSecondary.Mui-focused .MuiOutlinedInput-notchedOutline': {
          borderColor: theme.palette.error.main,
        },
      },
      '& .MuiInputAdornment-root > p': {
        color: theme.palette.common.black,
      },
    },
    '& .Mui-error.MuiFormLabel-colorSecondary.Mui-focused': {
      color: theme.palette.error.main,
    },
    '& .MuiFormHelperText-contained': {
      marginLeft: 0,
    },
  },
}));

const QuantityField: React.FC<QuantityFieldPropTypes> = ({
  name,
  label,
  defaultValue,
  onFocus,
  onBlur,
  error,
  errorText,
  shouldFocus = true,
  startAdornment,
}) => {
  const classes = useStyles();
  const inputRef = useRef<HTMLInputElement>(null);
  const [hasFocus, setHasFocus] = useState(false);

  useEffect(() => {
    if (shouldFocus) {
      inputRef.current.focus();
    }
  }, [shouldFocus]);

  useEffect(() => {
    if (hasFocus) {
      inputRef?.current?.select();
    }
  }, [hasFocus]);

  const onhandleFocus = () => {
    setHasFocus(true);
    if (onFocus) {
      onFocus();
    }
  };

  const onHandleBlur = (evt) => {
    setHasFocus(false);
    if (onBlur) {
      onBlur(evt);
    }
  };

  const hasError = !hasFocus && error;

  return (
    <TextField
      id={name}
      inputRef={inputRef}
      classes={classes}
      name={name}
      label={label}
      fullWidth
      inputProps={{ inputMode: 'numeric', pattern: '[0-9]*' }}
      variant="outlined"
      defaultValue={defaultValue}
      onFocus={onhandleFocus}
      onBlur={onHandleBlur}
      error={hasError}
      helperText={hasError && errorText}
      InputLabelProps={{ shrink: true }}
      InputProps={{
        startAdornment: startAdornment ? <InputAdornment position="start">{startAdornment}</InputAdornment> : null,
      }}
      color="secondary"
      autoComplete="off"
    />
  );
};

export { QuantityField };
