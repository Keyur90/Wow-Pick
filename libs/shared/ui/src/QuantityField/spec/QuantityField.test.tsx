import React from 'react';
import { render } from '@testing-library/react';
import { QuantityField } from '../QuantityField';

test('renders QuantityField', async () => {
  const { container } = render(
    <QuantityField
      name="suppliedQuantity"
      label={'Enter the supplied quantity'}
      defaultValue={10}
      onChange={() => null}
      error={false}
      errorText={''}
    />
  );

  expect(container).toMatchSnapshot();
});
