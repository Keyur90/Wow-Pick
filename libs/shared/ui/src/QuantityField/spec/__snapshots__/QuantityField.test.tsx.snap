// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders QuantityField 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-root-1 MuiFormControl-fullWidth"
  >
    <label
      class="MuiFormLabel-root MuiFormLabel-colorSecondary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-focused Mui-focused"
      data-shrink="true"
      for="suppliedQuantity"
      id="suppliedQuantity-label"
    >
      Enter the supplied quantity
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorSecondary MuiOutlinedInput-colorSecondary MuiInputBase-fullWidth Mui-focused Mui-focused MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        autocomplete="off"
        class="MuiInputBase-input MuiOutlinedInput-input"
        id="suppliedQuantity"
        inputmode="numeric"
        name="suppliedQuantity"
        pattern="[0-9]*"
        type="text"
        value="10"
      />
      <fieldset
        aria-hidden="true"
        class="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="PrivateNotchedOutline-legendLabelled-4 PrivateNotchedOutline-legendNotched-5"
        >
          <span>
            Enter the supplied quantity
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
